.github/actions/assign-dependabot-reviewer/assign-reviewers.mjs

```diff
@@ -0,0 +1,101 @@
+import {setFailed} from '@actions/core';
+import {reviewersMap} from "./config/reviewers.mjs";
+import github from "@actions/github";
+
+const GH_TOKEN = process.env.GH_TOKEN;
+
+const REPO = github.context.repo.repo;
+const REPO_OWNER = github.context.repo.owner;
+const octokit = github.getOctokit(GH_TOKEN)
+
+/**
+ * Gets the pull request number, if the event was from a comment, pull it from the issue number
+ * If the event was from a PR, pull it from the pull request number
+ * @returns {number} The PR number
+ */
+function getPRNumber() {
+    if (github.context.eventName === 'issue_comment') {
+        return github.context.payload.issue.number;
+    }
+
+    return github.context.payload.pull_request.number;
+}
+
+/**
+ * Gets the pull request title, if the event was from a comment, pull it from the issue
+ * If the event was from a PR, pull it from the pull request
+ * @returns {number} The PR number
+ */
+function getPRTitle() {
+    if (github.context.eventName === 'issue_comment') {
+        return github.context.payload.issue.title;
+    }
+
+    return github.context.payload.pull_request.title;
+}
+
+/**
+ * Assigns the PR to a team
+ * @param team_reviewers The teams to assign to
+ */
+async function assignReviewers(team_reviewers) {
+    await octokit.rest.pulls.requestReviewers({
+        owner: REPO_OWNER,
+        repo: REPO,
+        pull_number: getPRNumber(),
+        team_reviewers
+    });
+}
+
+/**
+ * Gets the dependency from the PR title using a regex to extract from the group or single dependency
+ * @param prTitle The PR title
+ * @returns {RegExp}
+ */
+function getDependencyNameRegex(prTitle) {
+    if (prTitle.includes('group')) {
+        return /the (\S+) group/;
+    }
+
+    return /Bump (\S+) from/
+}
+
+/**
+ * Extracts the dependency name from a PR title
+ * @param str
+ * @returns {null|string}
+ */
+function extractDependencyName(str) {
+  const match = str.match(getDependencyNameRegex(str));
+
+  // Return the first group or null if not found
+  return match ? (match[1]).trim() : null;
+}
+
+async function main() {
+    try {
+        const prTitle = getPRTitle();
+        // Determine the dependency using the title of the PR
+
+        const reviewers = [];
+        const dependencyName = extractDependencyName(prTitle);
+        const reviewer = reviewersMap[dependencyName];
+        console.log(`Found ${reviewer} for ${dependencyName}`)
+
+        if (reviewer != null) {
+            reviewers.push(reviewer);
+        }
+
+        if (reviewers.length > 0) {
+            await assignReviewers(reviewers);
+            console.log(`Successfully assigned reviewers: ${reviewers.join(', ')}`);
+        } else {
+            console.log('No reviewers to assign based on updated dependencies.');
```

Comment:
Should this be considered an error state? Should we notify a team that the reviewers map is outdated? Or is this better handled every couple of months via manual review?

Reply:
No - this shouldn't happen as platform will be the default maintainer as a fallback. The log message indicates the fallback to platform failed and to avoid having the PR show a "failed" check (failure to assign the PR shouldn't block the PR), I have this silently fail instead.